### PR TITLE
Update Cosmos SDK

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Refactor some tests

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -13,7 +13,7 @@
   <PropertyGroup Label="Dependencies from nuget.org">
     <SystemThreadingTasksExtensionsVersion>4.5.2</SystemThreadingTasksExtensionsVersion>
     <CastleCorePackageVersion>4.2.1</CastleCorePackageVersion>
-    <MicrosoftAzureCosmosPackageVersion>3.0.0.1-preview</MicrosoftAzureCosmosPackageVersion>
+    <MicrosoftAzureCosmosPackageVersion>3.0.0.17-preview</MicrosoftAzureCosmosPackageVersion>
```
Comment (dougbu) [inline eng/Versions.props]: @Pilchie does this new version need your approval?

Comment (Pilchie) [reply eng/Versions.props]: Is that an OSS dependency, or just an external package?

Comment (ajcvickers) [reply eng/Versions.props]: It's a first-party package shipped by Azure. I don't believe this has needed approval in the past.

Comment (Pilchie) [reply eng/Versions.props]: I wouldn't expect so.

**Pull request merged**